src/v2/recommender.py

```diff
@@ -4,474 +4,202 @@
-            package['cooccurrence_probability'] = float(100.0)
-        new_package_list.append(package)
-    return new_package_list
+ECOSYSTEM_TO_SERVICE_HOST = {
+    "pypi": _prepare_insights_url(Settings().pypi_service_host),
+    "npm": _prepare_insights_url(Settings().chester_service_host),
+    "maven": _prepare_insights_url(Settings().maven_service_host),
+}
 
 
 class RecommendationTask:
     """Recommendation task."""
 
-    _analysis_name = 'recommendation_v2'
-    description = 'Get Recommendation'
-    kronos_ecosystems = ['maven']
-    chester_ecosystems = ['npm']
-    hpf_ecosystems = ['maven']
-    pypi_ecosystems = ['pypi']
-    golang_ecosystem = ['golang']
-
-    @staticmethod
-    def get_insights_url(payload):
-        """Get the insights url based on the ecosystem."""
-        if payload and 'ecosystem' in payload[0]:
-            if payload[0]['ecosystem'] in RecommendationTask.chester_ecosystems:
-                INSIGHTS_SERVICE_HOST = os.getenv("CHESTER_SERVICE_HOST")
-            elif payload[0]['ecosystem'] in RecommendationTask.pypi_ecosystems:
-                INSIGHTS_SERVICE_HOST = os.getenv("PYPI_SERVICE_HOST")
-            elif payload[0]['ecosystem'] in RecommendationTask.golang_ecosystem:
-                INSIGHTS_SERVICE_HOST = os.environ.get("GOLANG_SERVICE_HOST")
-            else:
-                INSIGHTS_SERVICE_HOST = os.getenv("HPF_SERVICE_HOST") + "-" + payload[0][
-                    'ecosystem']
-
-            INSIGHTS_URL_REST = "http://{host}:{port}".format(host=INSIGHTS_SERVICE_HOST,
-                                                              port=os.getenv("SERVICE_PORT"))
-
-            insights_url = INSIGHTS_URL_REST + "/api/v1/companion_recommendation"
-
-            return insights_url
-
-        else:
-            logger.error('Payload information not passed in the call, Quitting! inights '
-                         'recommender\'s call')
-
     @staticmethod
-    def call_insights_recommender(payload):
+    def _call(payload: InsightsRequest):
         """Call the PGM model.
 
         Calls the PGM model with the normalized manifest information to get
         the relevant packages.
         """
+        insights_url = ECOSYSTEM_TO_SERVICE_HOST.get(payload.ecosystem, None)
+        assert insights_url
         try:
-            # TODO remove hardcodedness for payloads with multiple ecosystems
-
-            insights_url = RecommendationTask.get_insights_url(payload)
-            response = get_session_retry().post(insights_url, json=payload)
-
-            if response.status_code != 200:
-                logger.error("HTTP error {}. Error retrieving insights data.".format(
-                             response.status_code))
-                return None
-            else:
-                json_response = response.json()
-                return json_response
-
+            response = get_session_retry().post(insights_url, json=[payload.dict()])
```

Comment:
Can we add timestamp logs around insight service call?, this log will help in investigation of timeout errors (408).

Reply:
Done